holidays/countries/ireland.py

```diff
@@ -92,12 +92,14 @@ def _populate(self, year):
         name = "Christmas Day"
         self[date(year, DEC, 25)] = "Christmas Day"
         if self.observed and date(year, DEC, 25).weekday() in WEEKEND:
-            self[date(year, DEC, 27) + rd(weekday=MON)] = name + " (Observed)"
+            self[date(year, DEC, 25) + rd(weekday=MON)] = name + " (Observed)"
 
         # St. Stephen's Day
         name = "St. Stephen's Day"
         self[date(year, DEC, 26)] = name
-        if self.observed and date(year, DEC, 26).weekday() in WEEKEND:
+        if self.observed and date(year, DEC, 26).weekday() == SAT:
+            self[date(year, DEC, 26) + rd(weekday=MON)] = name + " (Observed)"
+        elif self.observed and date(year, DEC, 26).weekday() == SUN:
             self[date(year, DEC, 26) + rd(weekday=TUE)] = name + " (Observed)"
```

Comment:
Now I understand this part.
I think that could be correct as is, just that maybe I would use:
```python
        if self.observed and date(year, DEC, 26).weekday() in WEEKEND:
            self[date(year, DEC, 26) + rd(days=2)] = name + " (Observed)"
```
I'm not sure if it's more understandable or easier to maintain, so I'll leave it as a suggestion.

Reply:
I like it being a little bit more verbose ;)

Reply:
It's fine to me 👍🏻